base_phone_popup/models/phone_common.py

```diff
@@ -0,0 +1,122 @@
+        nr_digits_to_match_from_end = \
+            self.env.user.company_id.number_of_digits_to_match_from_end
+        if len(presented_number) >= nr_digits_to_match_from_end:
+            end_number_to_match = presented_number[
+                -nr_digits_to_match_from_end:len(presented_number)]
+        else:
+            end_number_to_match = presented_number
+        sorted_phonemodels = self._get_phone_models()
+        for obj_dict in sorted_phonemodels:
+            obj = obj_dict['object']
+            pg_search_number = str('%' + end_number_to_match)
+            _logger.debug(
+                "Will search phone and mobile numbers in %s ending with '%s'",
+                obj._name, end_number_to_match)
+            domain = []
+            for field in obj_dict['fields']:
+                domain.append((field, '=like', pg_search_number))
+            if len(domain) > 1:
+                domain = ['|'] * (len(domain) - 1) + domain
+            _logger.debug('searching on %s with domain=%s', obj._name, domain)
+            res_obj = obj.search(domain)
+            if len(res_obj) > 1:
+                _logger.warning(
+                    "There are several %s (IDS = %s) with a phone number "
+                    "ending with '%s'. Taking the first one.",
+                    obj._name, res_obj.ids, end_number_to_match)
+                res_obj = res_obj[0]
+            if res_obj:
+                name = res_obj.name_get()[0][1]
+                res = (obj._name, res_obj.id, name)
+                _logger.debug(
+                    "Answer get_record_from_phone_number: (%s, %d, %s)",
+                    res[0], res[1], res[2])
+                return res
+            else:
+                _logger.debug(
+                    "No match on %s for end of phone number '%s'",
+                    obj._name, end_number_to_match)
+        return False
+
+    @api.model
+    def _get_phone_models(self):
+        # Hook to get models and fields on which search
+        # You must inherit if you want to search into another fields
+        res = [
+            {'fields': ['phone', 'mobile'], 'object': self.env['res.partner']},
+            {'fields': ['phone', 'mobile'], 'object': self.env['crm.lead']},
+        ]
+        return res
+
+    @api.model
+    def _prepare_incall_pop_action(self, record_res, number):
+        if record_res:
+            obj = self.env[record_res[0]]
+            action = {
+                'name': obj._description,
+                'type': 'ir.actions.act_window',
+                'res_model': record_res[0],
+                'view_mode': 'form,tree',
+                'views': [[False, 'form']],  # Beurk, but needed
+                'target': 'new',
+                'res_id': record_res[1],
+                }
+        else:
+            action = {
+                'name': _('Number Not Found'),
+                'type': 'ir.actions.act_window',
+                'res_model': 'number.not.found',
```

Comment:
You're calling the "number.not.found" wizard defined in base_phone but you don't depend on base_phone. BTW, it's one of the things base_phone is useful for :)

Reply:
OK, I see. When are you going to publish `base_phone` to v12? So that we can backport it and use it as dependency for this (we will later return the favor migrating this to v12).

Reply:
I ported the main phone module to v12 (base_phone, asterisk_click2dial, crm_phone, hr_phone, hr_recruitment_phone, event_phone), starting from the v10 code. As I couldn't make a PR because the current v12 branch had nothing in common with my v12 branch coming from v10, I pushed it directly, and now we can improve/make PR, etc...

This morning, I made a quick backport of that work from v12 to v11. It's starting to work well; I should be able to finish the tests/cleanup in the coming days.
@pedrobaeza If you are OK with that, I will push my backport directly to origin/11.0 like I did for v12, and then we can make PRs on it. It may not be a standard OCA way of doing such things, but if we want to move forward on v11 for connector-telephony, I think it's the right way to move forward and unlock this project.

Reply:
@alexis-via you should do a PR, although you merge yourself when CIs are green and several days have passed, but at least do that for:

* not polluting main branch with several trials for greenifying CIs.
* let the opportunity to comment some things.

One thing you are not going to attend for sure is the commit squashing for reducing the useless ones, so let also that others like me do that task.

And please one PR per module.

Reply:
I won't do 1 PR per module, because they all depend one on each other, it's too much (useless) work. These modules work together, they must be tested together, so it's much better to put them in the same PR.

How do you make a PR if there is no common history? I tried to make a PR yesterday night for v12, but I couldn't find a way, that's why I decided to push directly.

Reply:
@alexis-via they don't depend each other. It's a hierarchy. Look how we are migrating OCA/bank-payment. About the PR, you just need to follow the procedure in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0, changing remotes if you are going to migrate from v12.

Reply:
@alexis-via if you need to try all together is because they are not enough decoupled or good unit testing. Any way, you are going to do what you want as always...

Reply:
Unit testing is difficult because you need an asterisk server to make real tests!